src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp

```diff
@@ -287,7 +354,28 @@ void FlowSolverBase::updatePorosityAndPermeability( CellElementSubRegion & subRe
   {
     typename TYPEOFREF( castedPorousSolid ) ::KernelWrapper porousWrapper = castedPorousSolid.createKernelUpdates();
 
-    execute1( porousWrapper, subRegion, pressure );
+    if( m_isThermal )
+    {
+      if( m_isBiot )
+      {
+        execute2( porousWrapper, subRegion, pressure, pressure_n, temperature, temperature_n );
+      }
+      else
+      {
+        execute2( porousWrapper, subRegion, pressure, temperature ); 
+      }
+    }
+    else
+    {
+      if( m_isBiot )
+      {
+        execute1( porousWrapper, subRegion, pressure, pressure_n );
+      }
+      else
+      {
+        execute1( porousWrapper, subRegion, pressure ); 
+      }
+    }
```

Comment:
> If possible, if I understood well what you do in the porosity models, we could arrive at something like this:
> 
> in two ways:
> 
> 1. Always pass the variable at the current time and the variable at the previous time. The models that do not need the variable at the previous time (like `PressurePorosity`) will not use it.
> 2. Always pass the delta pressure and the delta temperature (= 0 in isothermal simulations). The porosity update in the `BiotPorosity` will be able to handle that because the temperature-dependent update will be zero. The isothermal models like `PressurePorosity` will simply not use the temperatures.
> 
> With that, you will not need the `m_isBiot` anymore, which would be great!

Thanks for this suggestion! I've changed `updateStateFromPressure` and `updateStateFromPressureAndTemperature` such that they take both current and previous values of pressure and/or temperature as inputs. In the way, we don't need `m_isBiot`. I still keep two executes here as I want to separate the updates for the isothermal and non isothermal cases for now, given that we have `m_isThermal` flag here.